### PR TITLE
KAA-396: Update C SDK to support user verifiers.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: client/client-multi/client-c/src/kaa_user.c [hunk 1]
```diff
@@ -32,14 +34,19 @@
...
-    size_t  user_access_token_len;
-    char   *user_external_id;
-    char   *user_access_token;
+    char          *user_external_id;
+    size_t        user_external_id_len;
+    char          *user_access_token;
+    size_t        user_access_token_len;
+    char          *user_verifier_token;
+    size_t        user_verifier_token_len;
+    destroy_fn    destroy;
```
Comment (izeigerman) [inline client/client-multi/client-c/src/kaa_user.c]: Why do you need the destroy callback?

Comment (ghost) [inline client/client-multi/client-c/src/kaa_user.c]: To tell the truth, there is no any specific reason. It is a mini enterprise :)
I will get rid off this callback and replace it with direct call of the destroy function.

File: client/client-multi/client-c/src/kaa_user.c [hunk 2]
```diff
@@ -172,9 +219,11 @@ static size_t kaa_user_request_get_size_no_header(kaa_user_manager_t *self)
 {
     size_t expected_size = 0;
     if (self->user_info && !self->is_waiting_user_attach_response) {
-       expected_size += sizeof(uint32_t); // External system authentication field header
+       expected_size += sizeof(uint32_t)
```
Comment (ohamov) [inline client/client-multi/client-c/src/kaa_user.c]: add comment please

File: client/client-multi/client-c/src/kaa_user.c [hunk 3]
```diff
@@ -156,15 +192,26 @@ kaa_error_t kaa_user_manager_attach_to_user(kaa_user_manager_t *self
     return KAA_ERR_NONE;
 }
 
+kaa_error_t kaa_user_manager_default_attach_to_user(kaa_user_manager_t *self
+                                                  , const char *user_external_id
+                                                  , const char *access_token)
+{
+    KAA_RETURN_IF_NIL3(self, user_external_id, access_token, KAA_ERR_BADPARAM);
+
+    if (!strlen(DEFAULT_USER_VERIFIER_TOKEN)) {
```
Comment (izeigerman) [inline client/client-multi/client-c/src/kaa_user.c]: Compile-time verification is better idea.

**Comment by izeigerman** [comment]
LGTM

**Pull request merged**